scripts/run_project_tests.py

```diff
@@ -0,0 +1,141 @@
+    else:
+        result = subprocess.run(test_cmd, capture_output=True, text=True)
+
+    output = (result.stdout or result.stderr or "").strip()
+
+    if result.returncode != 0 or output.endswith("FAIL"):
+        print_fancy("FAIL", Fore.RED)
+        if not stream:
+            print_fancy("OUTPUT:", Fore.YELLOW)
+            print(output)
+    else:
+        print_fancy("OK", Fore.GREEN)
+
+    return result.returncode
+
+
+def parse_args() -> argparse.Namespace:
+    """Parses script arguments"""
+    parser = argparse.ArgumentParser(
+        description="Run tests with options:",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+    parser.add_argument("-C", "--campaign", default="ci", help="Set campaign to run.")
+    parser.add_argument("-c", "--config", default=".tests_config.yaml", help="Set path to config.")
+    parser.add_argument("-v", "--verbose", action="store_true", help="Enable verbose mode.")
+    parser.add_argument("-S", "--stream", action="store_true", help="Stream the output of test on stdout.")
+    return parser.parse_args()
+
+
+def get_target() -> str:
+    target = os.environ.get("TARGET")
+    if not target:
+        raise TestSetupError("TARGET environment variable is not set.")
+    return target
+
+
+def read_yaml_config(yaml_path: str) -> dict:
+    """Loads test configurations from YAML files"""
+    try:
+        with open(yaml_path, "r", encoding="utf-8") as file:
+            config = yaml.safe_load(file)
+    except Exception as e:
+        raise TestSetupError(f"Error while reading config file: {e}")
+
+    return config
+
+
+def run_tests(config: dict, campaign:str, target: str, verbose: bool, stream: bool) -> Dict[str, int]:
+    """Executes the specified tests"""
+    exit_codes = {}
+    for test_type, tests in config[campaign].items():
+        print_fancy(f"- Start {test_type} tests -", Fore.YELLOW)
+
+        for test in tests:
+            if test_type == "pytest":
+                test_cmd = ["pytest", test]
+                if not verbose:
+                    test_cmd.append("--tb=no")
+                if verbose or stream:
+                    test_cmd.append("-v")
+                test_cmd = [item for element in test_cmd for item in element.split()]
+            else:
+                test_cmd = [f"./_build/{target}/prog/{test}"]
+
+            exit_codes[test] = run_test(test_cmd, test, stream)
+
+            if exit_codes[test] != 0:
+                print_fancy(f"{test} exited with: {exit_codes[test]}", Fore.RED)
+
+    return exit_codes
+
+
+def main():
+    try:
+        args = parse_args()
+        target = get_target()
+        config = read_yaml_config(args.config)
+
+        completed_tests = run_tests(config, args.campaign, target, args.verbose, args.stream)
+        if any(code != 0 for code in completed_tests.values()):
+            if args.verbose:
+                print_fancy(f"Completed tests and their exit codes: {completed_tests}", Fore.RED)
```

Comment:
not sure it should be red (all failed test produce an error message in red inside `run_tests` already?)